omsApp/src/drvMAXv.cc

```diff
@@ -935,7 +930,7 @@ MAXvSetup(int num_cards,        /* maximum number of cards in rack */
             }
             else
             {
-                MAXv_addrs = (char *) addrs;
+                MAXv_addrs = (char *) (size_t) addrs;
```

Comment:
Are these double casts there to stop a compiler warning?

Reply:
> Which version of RTEMS was used for building?

I used RTEMS-4.9 but since there is nothing RTEMS specific here I guess it doesn't matter. All the architecture specific stuff is already nicely hidden in devLib.

> Was `motorIsrDisable` removed because it wasn't used?

Yes, it was an unused static routine and the compiler saw it was unused and issued a warning.

> The definition of USE_DEVLIB removed:
> 
> ```
> /* Are we using VME-Bus/devLib */
> #if (defined(vxWorks) || defined(__rtems__))
>     #define USE_DEVLIB
> #endif
> ```
> 
> This appears to have been used to avoid making VME-specific calls on non-VME platforms, since the `omsApp/src/Makefile` builds all the drivers for every platform. I would have expected this change to introduce problems when building for Linux.

Me too, but to my surprise it builds just fine on Linux. I think this is because the devLib has suitable dummy implementations for all methods on non-embedded OSes (i.e. everything except vxWorks and RTEMS).

> Are these double casts there to stop a compiler warning?

Yes, exactly. The warning was "cast to pointer from integer of different size".

Reply:
There is at least one site that has linux-x86 systems which connect to VME over a fibre connection. They have a devLib implementation that works with the Linux drivers for their hardware. Code that calls the devLib APIs has built on all our platforms for many years now; it actually only matters at runtime whether there is a real implementation behind the API. Thus if you have any code that uses devLib it can often be built on all architectures, although builds for Windows might need symbol decorations adding though.